ipc/cli/src/commands/subnet/create.rs

```diff
@@ -42,34 +42,23 @@ impl CreateSubnet {
             kind: arguments.supply_source_kind,
             token_address,
         };
-        let addr = provider
-            .create_subnet(
-                from,
-                parent,
-                arguments.min_validators,
-                f64_to_token_amount(arguments.min_validator_stake)?,
-                arguments.bottomup_check_period,
-                arguments
-                    .active_validators_limit
-                    .unwrap_or(DEFAULT_ACTIVE_VALIDATORS),
-                f64_to_token_amount(arguments.min_cross_msg_fee)?,
-                arguments.permission_mode,
-                supply_source,
-            )
-            .await?;
-
-        Ok(addr.to_string())
-    }
-}
 
-#[async_trait]
-impl CommandLineHandler for CreateSubnet {
-    type Arguments = CreateSubnetArgs;
+        let params = ConstructParams::new(
+            parent,
+            arguments.min_validators,
+            f64_to_token_amount(arguments.min_validator_stake)?,
+            arguments.bottomup_check_period,
+            f64_to_token_amount(arguments.min_cross_msg_fee)?,
+            arguments.permission_mode,
+            supply_source,
+        );
```

Comment:
There doesn't seem to be a way to set the non-default parameters. I'm wondering if it would be cleaner to make those fields non-optional, and pass the defaults to this method, so that it can fill them unconditionally, and then the caller has the option to overwrite them if they have the means to do so. I think it would be better than leaving them potentially empty and throwing errors in the final constructor transformation if you forgot to call this pre-flight method.

Reply:
Then we need a builder perhaps

Reply:
Your pattern of not-really-optional-but-optional-during-build is used in ethers, but that's because if not set, they can trigger all sorts of calculations to fill them in (nonce lookup, gas estimation, fee estimation). I could be wrong but our usecase seems simpler here.

Reply:
so a check is performed before sending to chain, I think I added into the data conversion. The setting optional part is not added yet as they are not exposed in `ipc-cli`.

Reply:
I know the check is there before sending to the chain. Sorry, what I was trying to say is that I think this is a bit convoluted, when all that happens is that the defaults will be set by this provider implementation or that provider implementation.. might as well be set when the record is created in `new` and then there is no need for checks or calling anything, _unless_ you want to override something. Since none of these fields require compilicated estimations _iff_ they are empty, I don't see what benefits leaving them as `None` has.

Reply:
Ok, I see where you are coming from. Yeah, I can do this too. I thought we might have to config those `Optional` parameter. But we can do those later too.